stl/inc/ranges

```diff
@@ -1467,22 +1460,30 @@ namespace ranges {
                 return _Last;
             }
 
-            _NODISCARD friend constexpr bool operator==(const _Iterator<_Const>& _Left,
-                const _Sentinel& _Right) noexcept(noexcept(_Right._Equal(_Left))) /* strengthened */ {
-                return _Right._Equal(_Left);
+            // clang-format off
+            template <bool _OtherConst>
+                requires sentinel_for<sentinel_t<_Base>, _Maybe_const_iter<_OtherConst>>
+            _NODISCARD friend constexpr bool operator==(const _Iterator<_OtherConst>& _Left,
+                const _Sentinel& _Right) noexcept(noexcept(_Get_current(_Left) == _Right._Last)) /* strengthened */ {
+                return _Get_current(_Left) == _Right._Last;
             }
 
-            _NODISCARD friend constexpr range_difference_t<_Base>
-                operator-(const _Iterator<_Const>& _Left, const _Sentinel& _Right) noexcept(
-                    noexcept(_Right._Distance_to(
-                        _Left))) /* strengthened */ requires sized_sentinel_for<sentinel_t<_Base>, iterator_t<_Base>> {
-                return _Right._Distance_to(_Left);
+            template <bool _OtherConst>
+                requires sized_sentinel_for<sentinel_t<_Base>, _Maybe_const_iter<_OtherConst>>
+            _NODISCARD friend constexpr range_difference_t<_Maybe_const<_OtherConst, _Vw>> operator-(
+                const _Iterator<_OtherConst>& _Left,
+                const _Sentinel& _Right) noexcept(noexcept(_Get_current(_Left) - _Right._Last)) /* strengthened */ {

```

Comment:
Suggestion: Adjust wrapping (in this non-clang-formatted region) to match the overload below (which simply reverses the parameters):
```suggestion
            _NODISCARD friend constexpr range_difference_t<_Maybe_const<_OtherConst, _Vw>>
                operator-(const _Iterator<_OtherConst>& _Left, const _Sentinel& _Right) noexcept(
                    noexcept(_Get_current(_Left) - _Right._Last)) /* strengthened */ {
```